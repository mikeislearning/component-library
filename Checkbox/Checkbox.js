import React from 'react';
import CheckBox from 'react-native-check-box';
import { View } from 'react-native';
import { PropTypes } from 'prop-types';
import { colors } from 'app/theme';

import styles from './Checkbox.styles';

export default class Checkbox extends React.Component {
  static propTypes = {
    isChecked: PropTypes.bool,
    onClick: PropTypes.func,
    rightText: PropTypes.string,
    rightTextView: PropTypes.node,
  };

  render() {
    const { isChecked, onClick, rightText, rightTextView } = this.props;

    return (
      <View>
        <CheckBox
          style={styles.checkbox}
          onClick={onClick}
          isChecked={isChecked}
          rightText={rightText}
          rightTextView={rightTextView}
          checkedCheckBoxColor={colors.green.dark}
          uncheckedCheckBoxColor={colors.gray.medium}
        />
      </View>
    );
  }
}
