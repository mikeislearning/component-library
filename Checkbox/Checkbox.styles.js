import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  checkbox: {
    flex: 1,
    paddingRight: 10,
  },
});

export default styles;
