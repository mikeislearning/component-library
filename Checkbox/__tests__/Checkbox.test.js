/* eslint-disable no-console */
/**
 * @format
 */

import React from 'react';
import { render, fireEvent } from 'react-native-testing-library';

import Checkbox from '../Checkbox';

describe('<Checkbox />', () => {
  const onEventMock = jest.fn();
  it('renders correctly', () => {
    const checkBox = render(<Checkbox testID="checkBox" isChecked={true} onClick={onEventMock} />);
    const { getByTestId } = checkBox;
    expect(getByTestId('checkBox').props.isChecked).toBe(true);
    fireEvent(getByTestId('checkBox'), 'click');
    expect(onEventMock).toHaveBeenCalled();
    expect(checkBox).toMatchSnapshot();
  });
});
