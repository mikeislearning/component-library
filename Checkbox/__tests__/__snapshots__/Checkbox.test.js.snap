// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> renders correctly 1`] = `
<View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flex": 1,
        "paddingRight": 10,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../react-native-check-box/img/ic_check_box.png",
          }
        }
        style={
          Object {
            "tintColor": "#00BB83",
          }
        }
      />
    </View>
  </View>
</View>
`;
